app/controllers/trainees/lead_schools_controller.rb

```diff
@@ -22,6 +27,10 @@ def update
 
   private
 
+    def redirect_to_search_page
+      redirect_to trainee_lead_schools_path(trainee, query: params["input-autocomplete"]) if trainee_params[:lead_school_id].blank?
```

Comment:
This will only work for trainees that haven't previously had a `lead_school_id` saved.

If you return to the form, turn off JS, submit a query, then the `lead_school_id` will be populated with the already-saved ID and you will skip to the confirm page.

Reply:
Good catch, I've now opted to set the `lead_school_id` to `nil` in the form, [in this commit](https://github.com/DFE-Digital/register-trainee-teachers/pull/800/commits/1ba2fc0e482d9b72939306f16c60ba9d0509dbc6), and added another test to validate that behaviour.
Happy to change to a js based approach if that's more preferable?

Reply:
But doesn't this mean the user's choice is now visibly lost when you return to the page?

Reply:
Added change to ensure that the user's choice is re-hydrated on returning to this page.

Reply:
Hmmm now the value in the input is rehydrated, but the autocomplete results are undefined:

<img width="775" alt="Screenshot 2021-04-29 at 16 49 03" src="https://user-images.githubusercontent.com/18436946/116580993-cfbdf800-a90b-11eb-85e7-5770f9f9b5a2.png">

Reply:
Seems to be an open issue raised at https://github.com/alphagov/accessible-autocomplete/issues/424